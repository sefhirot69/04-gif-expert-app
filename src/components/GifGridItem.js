import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({id, img, title}) => {

    return (
        <div key={id} className='card animate__animated animate__fadeIn animate__delay-1s'>
            <img src={ img } alt={title}/>
            <p>{title}</p>
        </div>
    );
};

GifGridItem.propTypes = {
    id : PropTypes.string.isRequired,
    img : PropTypes.string.isRequired,
    title : PropTypes.string.isRequired
};

export default GifGridItem;