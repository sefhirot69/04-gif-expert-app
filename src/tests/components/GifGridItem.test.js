import {shallow} from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas sobre el componente GidGridItem', () => {

    test('debe mostrar el componente correctamente', () => {

        const args = {
            id : 1,
            img : 'https://google.es',
            title : 'titulo'
        };

        const wrapper = shallow(<GifGridItem {...args}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').text().trim()).toBe(args.title);
    });
});