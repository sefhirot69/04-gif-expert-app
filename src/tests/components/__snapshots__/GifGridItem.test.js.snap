// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GidGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
  key="1"
>
  <img
    alt="titulo"
    src="https://google.es"
  />
  <p>
    titulo
  </p>
</div>
`;
